pkg/fakerp/server.go

```diff
@@ -57,57 +61,22 @@ func NewServer(log *logrus.Entry, resourceGroup, address string) *Server {
 	return s
 }
 
-func (s *Server) ListenAndServe() {
-	// TODO: match the request path the real RP would use
-	http.Handle("/", s)
-	http.Handle("/admin", s)
-	httpServer := &http.Server{Addr: s.address}
-	s.log.Infof("starting server on %s", s.address)
-	s.log.WithError(httpServer.ListenAndServe()).Warn("Server exited.")
-}
-
-// ServeHTTP handles an incoming request to the server.
-func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	defer req.Body.Close()
-
-	// validate the request
-	ok := s.validate(w, req)
-	if !ok {
-		return
-	}
+func (s *Server) Run() {
+	s.router.Use(middleware.DefaultCompress)
+	s.router.Use(middleware.RedirectSlashes)
+	s.router.Use(middleware.Recoverer)
+	s.router.Use(s.logger)
+	s.router.Use(s.validator)
 
-	// process the request
-	switch req.Method {
-	case http.MethodDelete:
-		s.handleDelete(w, req)
-	case http.MethodGet:
-		s.handleGet(w, req)
-	case http.MethodPut:
-		s.handlePut(w, req)
-	}
-}
+	s.router.Delete("/", s.handleDelete)
+	s.router.Get("/", s.handleGet)
+	s.router.Put("/", s.handlePut)
```

Comment:
These now can be more gradually split to handler functions. we should not pass the request from admin and customer API to same handler function now.

Reply:
yes those can be done in a series of follow-ups.

Reply:
You *may* want to use different entrypoints but you definitely want to share as much logic as possible.